Installation Instructions, Step 1 (in-tree llvm build) does not work

The installation instructions say:

> The first step is to compile LLVM on your machine (this assumes an in-tree build, but out-of-tree will also work):
> 
> ```
>  cd $HOME
>  git clone https://github.com/llvm-mirror/llvm
>  cd llvm
>  git checkout release_37
>  ./configure
>  make
> ```

`./configure` results in:
`configure: error: In-source builds are not allowed. Please configure from a separate build directory!`

Please update instructions.

Step 3 also does not work:

> ```
> gcc -o main main.cbe.c
> main.cbe.c:8:21: fatal error: APInt-C.h: No such file or directory
>  #include <APInt-C.h>
>                      ^
> ```

`APInt-C.h` can be found at https://github.com/JuliaLang/julia/tree/master/src, but one can manually delete the offending line in `main.cbe.c` (at least in this case).

Also, the selection sort example is not in `llvm/lib/test/selectionsort`, it can be found in `llvm/projects/llvm-cbe/test/selectionsort`.

--- a/README.md
+++ b/README.md
@@ -32,7 +32,9 @@ The first step is to compile LLVM on your machine
      git clone https://github.com/llvm-mirror/llvm
      cd llvm
      git checkout release_37
-     ./configure
+     mkdir build
+     cd build
+     ../configure
      make
 
 Step 2: Compiling LLVM-CBE
@@ -41,8 +43,8 @@ Step 2: Compiling LLVM-CBE
 Next, download and compile llvm-cbe from the same folder:
 
     cd $HOME/llvm/projects
-    git clone https://github.com/JuliaComputing/llvm-cbe.git llvm-cbe
-    cd ..
+    git clone https://github.com/JuliaComputing/llvm-cbe
+    cd ../build
     make
 
 Step 3: Usage Examples
@@ -50,17 +52,18 @@ Step 3: Usage Examples
 
 If llvm-cbe compiles, you should be able to run it with the following commands.
 ```
-$ cd llvm/lib/test/selectionsort
+$ cd $HOME/llvm/projects/llvm-cbe/test/selectionsort
 $ ls
 main.c
 $ clang -S -emit-llvm main.c
 $ ls
 main.c main.ll
-$ $(HOME)/llvm/Release/bin/llvm-cbe main.ll
-
+$ $(HOME)/llvm/build/Debug+Asserts/bin/llvm-cbe main.ll
 ```
+
 Compile Generated C-Code and Run
 ================================
+
 ```
 $ gcc -o main.cbe main.cbe.c
 $ ls

